middleware/middleware.go

```diff
@@ -0,0 +1,70 @@
+package middleware
+
+import "net/http"
+
+const (
+	// HTTP Header
+	REQUEST_ID_HEADER = "X-Request-Id"
+	USER_ID_HEADER    = "X-FOULKON-USER-ID"
+
+	// Middleware names
+	AUTHENTICATOR_MIDDLEWARE  = "AUTHENTICATOR"
+	XREQUESTID_MIDDLEWARE     = "XREQUESTID"
+	REQUEST_LOGGER_MIDDLEWARE = "REQUEST-LOGGER"
+)
+
+// MiddlewareHandler is handler with a middleware list to apply
```

Comment:
MiddlewareHandler handles the HTTP request and applies its list of middlewares before calling the API